experiments/scripts/demoChecker.sh

```diff
@@ -0,0 +1,78 @@
+#!/bin/bash
+
+# Default values
+INPUT_DIR_COUNT=0
+INPUT_DIRS=()
+OUTPUT_DIR=$HOME/.demoChecker
+TIMEOUT_DURATION=30
+TIMEOUT_KILL_DURATION=60
+
+# Command line arguments
+for arg in "$@"
+do
+    case $arg in
+        -c|--input-count)
+            INPUT_DIR_COUNT="$2"
+            shift
+            shift
+        ;;
+        -dirs|--directories)
+            shift
+            for counter in $(seq 1 $INPUT_DIR_COUNT)
+            do
+                INPUT_DIRS+=("$1")
+                shift
+            done
+        ;;
+        -o|--output)
+            OUTPUT_DIR="$2"
+            shift
+            shift
+        ;;
+        -t|--timout-duration)
+            TIMEOUT_DURATION="$2"
+            shift
+            shift
+        ;;
+        -tk|--timeout-kill-duration)
+            TIMEOUT_KILL_DURATION="$2"
+            shift
+            shift
+        ;;
+    esac
+done
+
+# Prepare pass and fail files
+if [ ! -d $OUTPUT_DIR ]; then
+    mkdir -p $OUTPUT_DIR
+fi
+if [ -f "$OUTPUT_DIR/passFile" ]; then
+    rm "$OUTPUT_DIR/passFile"
+    touch "$OUTPUT_DIR/passFile"
+fi
+if [ -f "$OUTPUT_DIR/failFile" ]; then
+    rm "$OUTPUT_DIR/failFile"
+    touch "$OUTPUT_DIR/failFile"
+fi
+
+echo "Output Directory: " $OUTPUT_DIR
+
+# Run through the executables in the INPUT_DIRS
+for dir in "${INPUT_DIRS[@]}"
+do
+    echo "Dir: " $dir
+    pushd $dir
+    for file in $(find . -executable -type f)
+    do
+        echo "File: " $file
+        timeout -k $TIMEOUT_KILL_DURATION $TIMEOUT_DURATION $file
+        returnCode=$?
+        if [ $returnCode -eq 0 ] || [ $returnCode -eq 124 ] || [ $returnCode -eq 137 ]; then
```

Comment:
Hmmm, it is a bit tricky here.

Can we assume all demo apps run indefinitely, until we stop them?
If we can assume this, then the only "pass" case would be for the app to timeout: if the apps terminates earlier (even with code 0), then it means that it did not run properly.

In any case, it's a good idea to store the return code alongside the file name. So I would transform this into:

```
# We expect apps to timeout, any other behavior is a failure
if [ $returnCode -eq 124 ] || [ $returnCode -eq 137 ] ; then
    echo $file $returnCode >> "$OUTPUT_DIR/passFile"
else
    echo $file $returnCode >> "$OUTPUT_DIR/failFile"
fi
```

Reply:
I think we can assume that. Out of all the demos I've seen so far, the only ones that finished early were the Vulkan Compute demos. And even they requried a final confirmation to fully terminate.